src/librustc/dep_graph/graph.rs

```diff
@@ -941,7 +951,7 @@ struct DepNodeData {
 
 pub(super) struct CurrentDepGraph {
     data: Lock<IndexVec<DepNodeIndex, DepNodeData>>,
-    node_to_node_index: Lock<FxHashMap<DepNode, DepNodeIndex>>,
+    node_to_node_index: Sharded<FxHashMap<DepNode, DepNodeIndex>>,
```

Comment:
Ah, I guess the motivation for making the locks more fine-grained is to be able to shared this map. Seems reasonable.